src/mbgl/shaders/mtl/background.cpp

```diff
@@ -0,0 +1,17 @@
+#include <mbgl/shaders/mtl/background.hpp>
+#include <mbgl/shaders/background_layer_ubo.hpp>
+
+namespace mbgl {
+namespace shaders {
+
+const std::array<AttributeInfo,1> ShaderSource<BuiltIn::BackgroundShader, gfx::Backend::Type::Metal>::attributes = {
+    AttributeInfo{0, gfx::AttributeDataType::Float3, 1, "a_pos"},
+};
```

Comment:
What is the motivation behind moving these to source files?

Reply:
I got duplicate symbol errors when they were in the headers.